datafusion/expr/src/function.rs

```diff
@@ -221,17 +221,17 @@ pub fn return_type(
         BuiltinScalarFunction::Concat => Ok(Utf8),
         BuiltinScalarFunction::ConcatWithSeparator => Ok(Utf8),
         BuiltinScalarFunction::DatePart => Ok(Float64),
-        BuiltinScalarFunction::DateTrunc | BuiltinScalarFunction::DateBin => {
-            match input_expr_types[1] {
-                Timestamp(Nanosecond, _) | Utf8 => Ok(Timestamp(Nanosecond, None)),
-                Timestamp(Microsecond, _) => Ok(Timestamp(Microsecond, None)),
-                Timestamp(Millisecond, _) => Ok(Timestamp(Millisecond, None)),
-                Timestamp(Second, _) => Ok(Timestamp(Second, None)),
-                _ => Err(DataFusionError::Internal(format!(
-                    "The {fun} function can only accept timestamp as the second arg."
-                ))),
-            }
-        }
+        // DateTrunc always makes nanosecond timestamps
+        BuiltinScalarFunction::DateTrunc => Ok(Timestamp(Nanosecond, None)),
```

Comment:
I guess that it is before because the second argument in `DateTrunc` signature is `Timestamp(Nanosecond, None)`. So it doesn't cause issue before?

Reply:
> So it doesn't cause issue before?

Because it's hidden by other BUG #6613 (from_plan create projection use original schema)